File: parser/udp.py
from scapy.layers.inet import UDP
from scapy.packet import Packet as ScapyPacket

from parser.packet import Packet


def parse_udp(p: Packet, raw: ScapyPacket) -> None:
    if not raw.haslayer(UDP):
        return

    udp = raw[UDP]
    p.protocol = "UDP"
    p.src_port = udp.sport
    p.dst_port = udp.dport
    p.payload = bytes(udp.payload)

File: parser/udp.go
package parser

import (
	"github.com/google/gopacket"
	"github.com/google/gopacket/layers"
)

func parseUDP(p *Packet, raw gopacket.Packet) {
	udpLayer := raw.Layer(layers.LayerTypeUDP)
	if udpLayer == nil {
		return
	}
	udp := udpLayer.(*layers.UDP)

	p.Protocol = "UDP"
	p.SrcPort = uint16(udp.SrcPort)
	p.DstPort = uint16(udp.DstPort)
	p.Payload = udp.Payload
}
